### ISSUE TITLE
Format of Picture::data not clear

### ISSUE THREAD

**Issue by Ploppz** [jameshurst/rust-id3#23]
If I want to read an image file and construct a `Picture` with its contents, it is not clear from the [documentation](https://docs.rs/id3/0.2.3/id3/frame/struct.Picture.html) how I should lay out the bytes of the `data` field. What format is assumed?

**Comment by polyfloyd** [comment]
Hey,

`data` is a buffer containing an image compressed using the file format described by `mime_type`. The usual file types are `image/jpeg` and `image/png`.

This is useful information to add to the documentation, I'll leave this ticket open until that's fixed.

**Comment by Ploppz** [comment]
Ah, so just the encoded contents of a jpeg or png file. Thanks.

**Comment by Ploppz** [comment]
The following may be related to this:
I add a jpeg image as cover art as follows
```rust
    let mut response = reqwest::get(&video.thumbdefault)?;
    if response.status().is_success() {
        let mut content: Vec<u8> = Vec::new();
        response.read_to_end(&mut content)?;
        let mime = "image/jpeg".to_string();
        tag.add_picture(id3::frame::Picture {
            mime_type: mime,
            picture_type: id3::frame::PictureType::CoverFront,
            description: "Youtube thumbnail".to_string(),
            data: content,
        });
    }
```

When I play it with `mpv` it says 
```
[ffmpeg/video] mjpeg: EOI missing, emulating
[ffmpeg/demuxer] mp3: Estimating duration from bitrate, this may be inaccurate
 (+) Video --vid=1 [P] 'Youtube thumbnail' (mjpeg 1280x720)
 (+) Audio --aid=1 (mp3 2ch 48000Hz)
Displaying attached picture. Use --no-audio-display to prevent this.
[ffmpeg/video] mjpeg: overread 8
[ffmpeg/video] mjpeg: EOI missing, emulating
```
And displays the 'cover art' picture in a window while playing the audio.  https://i.imgur.com/YoIcdCP.png . However, as you can see, there is a small green bar in the bottom right corner.
I'm not sure whether I am still putting the data incorrectly in the `Picture`?

**Comment by polyfloyd** [comment]
Ok, after some debugging I found out that the problem does not occur when the `Unsynchronisation` feature is disabled.

To encode a tag without this feature, build a custom encoder and use the `Encoder::encode_to_path` function which I've just introduced.

I'll do some debugging to see what is causing this.

**Comment by Ploppz** [comment]
Thanks for such a rapid fix! Works well indeed.

**Comment by polyfloyd** [comment]
Hey @Ploppz, the problem with the Unsynch decoder has been fixed in version 0.2.5.
